stock_picking_invoicing/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,529 @@
+        line_obj = self.env["account.move.line"]
+        values = line_obj.default_get(line_obj.fields_get().keys())
+        values.update(
+            {
+                "name": name,
+                "product_id": product.id,
+                "quantity": quantity,
+                "price_unit": price,
+                "move_line_ids": move_line_ids,
+                "move_id": invoice.id,
+            }
+        )
+        values = self._simulate_invoice_line_onchange(values, price_unit=price)
+        values.update({"name": name})
+        return values
+
+    def _update_picking_invoice_status(self, pickings):
+        """
+        Update invoice_state on pickings
+        :param pickings: stock.picking recordset
+        :return: stock.picking recordset
+        """
+        return pickings._set_as_invoiced()
+
+    def ungroup_moves(self, grouped_moves_list):
+        """Ungroup your moves, split them again, grouping by
+        fiscal position, max itens per invoice and etc
+        :param grouped_moves_list:
+        :return: list of grouped moves list
+        """
+        return [grouped_moves_list]
+
+    def _create_invoice(self, invoice_values):
+        """Override this method if you need to change any values of the
+        invoice and the lines before the invoice creation
+        :param invoice_values: dict with the invoice and its lines
+        :return: invoice
+        """
+        return self.env["account.move"].create(invoice_values)
+
+    def _action_generate_invoices(self):
+        """
+        Action to generate invoices based on pickings
+        :return: account.move recordset
+        """
+        pickings = self._load_pickings()
+        company = pickings.mapped("company_id")
+        if company and company != self.env.company:
+            raise UserError(_("All pickings are not related to your company!"))
+        pick_list = self._group_pickings(pickings)
+        invoices = self.env["account.move"].browse()
+        for pickings in pick_list:
+            moves = pickings.mapped("move_ids")
+            grouped_moves_list = self._group_moves(moves)
+            parts = self.ungroup_moves(grouped_moves_list)
+            for moves_list in parts:
+                invoice, invoice_values = self._build_invoice_values_from_pickings(
+                    pickings
+                )
+                lines = [(5, 0, {})]
+                line_values = False
+                for moves in moves_list:
+                    line_values = self._get_invoice_line_values(
+                        moves, invoice_values, invoice
+                    )
+                    if line_values:
+                        lines.append((0, 0, line_values))
+                if line_values:  # Only create the invoice if it has lines
+                    invoice_values["invoice_line_ids"] = lines
+                    invoice_values["invoice_date"] = self.invoice_date
```

Comment:
Having both line_ids and invoice_line_ids causes a AssertionError when sanitizing the invoice dict.
```suggestion
                    invoice_values["invoice_date"] = self.invoice_date
                    invoice_values["line_ids"] = []
```

Reply:
Change made as susgest, thanks